### PR TITLE
Fix the serialization error by ff73c7a

### PR DESCRIPTION
**Pull request by Rui-Minetest**

### CODE REVIEW

File: src/nodedef.cpp [hunk 1]
```diff
@@ -254,11 +254,14 @@ static void serializeSimpleSoundSpec(const SimpleSoundSpec &ss,
 	writeF1000(os, ss.gain);
 	writeF1000(os, ss.pitch);
```
Comment (sfan5) [inline src/nodedef.cpp]: You need similar code here for the case of new server and old client

**Comment by Rui-Minetest** [comment]
Updated.

File: src/nodedef.cpp [hunk 2]
```diff
@@ -462,9 +466,9 @@ void ContentFeatures::serialize(std::ostream &os, u16 protocol_version) const
 	collision_box.serialize(os, protocol_version);
 
 	// sound
-	serializeSimpleSoundSpec(sound_footstep, os);
-	serializeSimpleSoundSpec(sound_dig, os);
-	serializeSimpleSoundSpec(sound_dug, os);
+	serializeSimpleSoundSpec(sound_footstep, os, protocol_version);
+	serializeSimpleSoundSpec(sound_dig, os, protocol_version);
+	serializeSimpleSoundSpec(sound_dug, os, protocol_version);
```
Comment (sfan5) [inline src/nodedef.cpp]: no need for a protocol bump, this can use the content features version too (see line 393)

File: src/nodedef.cpp [hunk 3]
```diff
@@ -386,7 +390,7 @@ void ContentFeatures::serialize(std::ostream &os, u16 protocol_version) const
 	}
 
 	// version
-	writeU8(os, 10);
+	writeU8(os, 11);
```
Comment (sfan5) [inline src/nodedef.cpp]: this should be either 10 or 11 depending on the protocol version

File: src/nodedef.cpp [hunk 4]
```diff
@@ -462,9 +466,9 @@ void ContentFeatures::serialize(std::ostream &os, u16 protocol_version) const
 	collision_box.serialize(os, protocol_version);
 
 	// sound
-	serializeSimpleSoundSpec(sound_footstep, os);
-	serializeSimpleSoundSpec(sound_dig, os);
-	serializeSimpleSoundSpec(sound_dug, os);
+	serializeSimpleSoundSpec(sound_footstep, os, protocol_version);
```
Comment (sfan5) [inline src/nodedef.cpp]: and this should rely on the version from above instead of the protocol version

**Comment by Rui-Minetest** [comment]
Updated.

**Comment by nerzhul** [comment]
Tested, okay, thanks for your reactivity

**Pull request merged**